LandsitesTools/prepare_nlp_cases.py

```diff
@@ -1 +1,282 @@
+    
+    ### Check if the provided config path and file exist
+    try:
+        if not Path(args.cfg_path).is_dir():
+            raise ValueError("Path does not exist.")
+        
+        ### Save full path to cfg file
+        cfg_file_path = Path(args.cfg_path) / args.cfg_file
+        if not cfg_file_path.is_file():
+            raise ValueError("File does not exist at specified path.")
+    
+    except ValueError:
+        print("Error in command line input!")
+        raise
+        
+    ### Load settings file
+    settings = InterfaceSettings(cfg_file_path)
+    print(f"\nTrying to prepare sites specified in cfg file {args.cfg_file}...\n{settings.sites2run}")
+    cases_to_build = settings.sites2run
+    #print(f"\nsites specified in cfg file:\n{settings.parser['user']['sites2run']}")
+
+########################################################################################################
+########################################### Interactive mode ###########################################
+########################################################################################################
+        
+else:
+    ##### Start by printing all available cases to the console #####
+    
+    ### Read site info json file
+    try:
+        with open(Path("../config/sites.json"), "r") as json_file:
+            cases_dict = json.load(json_file) # use `json.loads` to do the reverse
+    except Exception as error:
+        print("Error when reading the site information json file (../config/sites.json)!")
+        raise error
+        
+    print("\n")
+    print("*********************** Available cases ***********************")
+    
+    ### Function to print site info in table format
+    def print_table_row(ind, name, res, lat, lon):
+        print('|','%-7s' % str(ind), '|', '%-10s' % str(name), '|', '%-10s' % str(res),
+              '|', '%-10s' % str(lat), '|', '%-10s' % str(lon),'|')
+    
+    ### Print table header...
+    print('---------------------------------------------------------------')
+    print_table_row("Index", "Name", "Resolution", "Lat.", "Lon.")
+    print('---------------------------------------------------------------')
+    
+    ### Store results also in DataFrame
+    cases_df = pd.DataFrame(columns=["name","res","lat","lon"])
+    
+    ### Loop through keys in site dictionary...
+    for idx,case in enumerate(cases_dict.keys()):
+        print_table_row(idx, cases_dict.get(case).get("name"), cases_dict.get(case).get("res"),
+                        cases_dict.get(case).get("lat"), cases_dict.get(case).get("lon"))
+        # Store in df
+        cases_df.loc[idx,"name"] = cases_dict.get(case).get("name")
+        cases_df.loc[idx,"res"] = cases_dict.get(case).get("res")
+        cases_df.loc[idx,"lat"] = cases_dict.get(case).get("lat")
+        cases_df.loc[idx,"lon"] = cases_dict.get(case).get("lon")
+        
+    print('---------------------------------------------------------------')
```

Comment:
It looks like you are building the same pandas.DataFrame that is loaded in the [InterfaceSettings class](https://github.com/NorESMhub/NorESM_LandSites_Platform/blob/platform_dev/LandsitesTools/interface_settings.py#L39-L40).
If the default settings file were read independently of the workflow (interactive/non-interactive), this DataFrame would be already loaded from the [path specified in the default settings](https://github.com/NorESMhub/NorESM_LandSites_Platform/blob/platform_dev/settings.txt#L26): for example you could move lines [46-61](https://github.com/lasseke/NorESM_LandSites_Platform/blob/dev_cases_py/LandsitesTools/prepare_nlp_cases.py#L46-L61) before [the "if" statement on line 44](https://github.com/lasseke/NorESM_LandSites_Platform/blob/dev_cases_py/LandsitesTools/prepare_nlp_cases.py#L44)

Reply:
Good observation! The point here was to only access the attributes that are needed for creating cases directly from the JSON file. My intention was to prevent unwanted behaviour if nested site attributes were to be added to the JSON at a later stage (and then using read_json() in [your code](https://github.com/NorESMhub/NorESM_LandSites_Platform/blob/platform_dev/LandsitesTools/interface_settings.py#L39-L40)). But agreeing on a centralized way forward if of course the better choice!

Reply:
I see your point. Once we have a more complete overview of site attributes, we can evaluate whether pandas.DataFrame is the right data structure: it supports attributes assigned only to some sites (for the rest of the sites, the missing field will be NaN), but it may not support nested attributes as you point out.